### PR TITLE
Add two events: `ExecutionEngine.ContextLoaded` and `ExecutionEngine.ContextUnloaded`

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

**Comment by shargon** [comment]
You readed my mind xD
https://github.com/neo-project/neo-vm/compare/master...shargon:events?expand=1&title=Events

**Comment by shargon** [comment]
I would like to add `OnSysCall` and `OnInstruction` too

**Comment by erikzhang** [comment]
> I would like to add OnSysCall and OnInstruction too

Where will they be used?

File: src/neo-vm/ExecutionEngine.cs [hunk 1]
```diff
@@ -62,6 +62,13 @@ public class ExecutionEngine : IDisposable
         public ExecutionContext EntryContext => InvocationStack.Count > 0 ? InvocationStack.Peek(InvocationStack.Count - 1) : null;
         public VMState State { get; internal protected set; } = VMState.BREAK;
 
+        #region Events
+        
+        public event EventHandler<ExecutionContext> ContextLoaded;
```
Comment (shargon) [inline src/neo-vm/ExecutionEngine.cs]: For events, I prefer to add `On` before the name

**Comment by shargon** [comment]
Only for debug purposes, we can log all the flow easier with this event

**Comment by erikzhang** [comment]
For debug purposes, we can use the `virtual` methods.

File: src/neo-vm/ExecutionEngine.cs (hunk 1 above)
Comment (erikzhang) [reply src/neo-vm/ExecutionEngine.cs]: It is no longer popular to add `On` before event names.

**Pull request merged**